### PR TITLE
#150971111 DRY out request validation

### PR DESCRIPTION
**Pull request by andela-ojoloko**
#### What does this PR do?
DRY out the check for if request is a valid JSON and the check for valid request parameters
#### Description of Task to be completed?
- DRY out check for valid json in request.
- DRY out check for field existence in request.
- update tests 
#### How should this be manually tested?
$pytest -v
#### What are the relevant pivotal tracker stories?
#150971111

### CODE REVIEW

File: api/helpers.py [hunk 1]
```diff
@@ -94,3 +96,46 @@ def patch_vote(**kwargs):
         new_vote.save()
         response = {"message": "Vote has been successfully registered"}, 201
     return response
+
+
+def validate_request(*expected_args):
+    """ This method validates Request JSON.
+    """
+
+    def Real_validate_request(f):
```
Comment (LarryWachira) [inline api/helpers.py]: Function name isn't very pythonic. See [this](https://www.python.org/dev/peps/pep-0008/#id45).

File: api/helpers.py [hunk 2]
```diff
@@ -94,3 +96,46 @@ def patch_vote(**kwargs):
...
+                return {"status": "fail",
+                        "data": {"message": "Request must be a valid JSON"}
+                       }, 400
+            payload = request.get_json()
+            if payload:
+                for values in expected_args:
+                    for value in values:
+                        if value == values[0]:
+                            continue
+                        if (value is "tags" and value in payload) and not validate_type(payload[value], values[0]):
```
Comment (LarryWachira) [inline api/helpers.py]: `==` should be used here instead of `is`. See [here](https://dbader.org/blog/difference-between-is-and-equals-in-python)

**Comment by LarryWachira** [comment]
Nice work on the decorator @andela-ojoloko !!

File: api/helpers.py (hunk 1 above)
Comment (andela-ojoloko) [reply api/helpers.py]: fixed

**Comment by LarryWachira** [comment]
LGTM

**Comment by LarryWachira** [comment]
Please rebase @andela-ojoloko so we can get this merged as soon as possible.

**Comment by andela-oowonikoko** [comment]
LGTM

**Comment by andela-milesanmi** [comment]
LGTM

**Pull request merged**